Commonalities/documentation/Working/user-consent.md

```diff
@@ -0,0 +1,405 @@
+|`consent` | This level applies to purposes which are unrelated enough to the service. | Opt-in purpose|
+|`terms_and_conditions`| The operations or tasks done by an application to provide its functionality, as described in the Terms and Conditions. | Opt-in purpose|
+
+
+> _NOTE: This list of levels MAY be extended with new values. The OpenID Provider (Auth Server) and the APIs used by the Relying Parties (client Applications) MUST be ready to support new values in the future._
+
+### Purpose relationship with other key concepts
+
+The key concepts defined to handle user personal information are related to purpose concept in the following way:
+
+![Key concepts relationship](./images/key-concepts-relationship.png)
+
+- Scopes tell what resources (e.g. APIs) are accessible under those scopes.
+- A purpose has associated a set of personal information scopes (PI scopes). 
+  - These are scopes that refer to resources providing access to Personal Information and thus requiring special protection as stated by e.g. the GDPR law.
+  - Getting access to PI scopes MUST always be done by explicitly declaring a purpose.
+- Scopes are assigned to an application, granting access to APIs and other resources.
+- Purposes are assigned to an application.
+
+![Purpose association to App](./images/purpose-association-to-app.png)
+
+Therefore, an application will have access to:
+  - APIs or resources covered by the assigned scopes
+  - APIs or resources providing personal information under the assigned purpose (because of the PI scopes included in that purpose)
+    - **Only** if user has granted her consent for that purpose
+    - Which may happen automatically (automatic purposes) or may need an explicit user action to provide consent (depending on the purpose level of the purpose associated to that consent). In the second case, the consent is created and stored and will be validated during authorization.
+
+## Using purpose parameter in the authorization request
+
+The application can request a list of purposes in the authorization request of the [OAuth 2.0 Authorization Framework](https://www.rfc-editor.org/rfc/rfc6749) using an additional `purpose` parameter:
+
+| **Authorize request param** | Description |
+| --- | --- |
+| `purpose` | The value of this parameter is expressed as a list of space-delimited, case-sensitive strings. The strings will be the id of the purposes. |
+
+This parameter has been defined in these guidelines to allow applications to provide a purpose/s name which represents the reason why a client application needs to process a certain piece of user Personal Information. Note that, as explained before, a purpose maps to a list of scopes giving access to Personal Information resources.
+
+_Example: requesting "purpose\_1", "purpose\_2" and "purpose\_3"_
+
+```
+POST /bc-authorize HTTP/1.1
+Authorization: Basic {Credentials}
+Content-Type: application/x-www-form-urlencoded
+
+purpose=purpose_1%20purpose_2%20purpose_3&
```

Comment:
Is space the best separator, as per some other reference? Or maybe comma?

Reply:
It follows OAuth 2.0 standard. https://www.rfc-editor.org/rfc/rfc6749
```The value of the scope parameter is expressed as a list of space-delimited, case-sensitive strings.```

Reply:
Great, thanks.